From: Michal Kalderon <Michal.Kalderon@cavium.com>
Date: Mon, 3 Jul 2017 21:55:25 +0300
Subject: qed: initialize ll2_syn_handle at start of function
Patch-mainline: v4.13-rc1
Git-commit: 25f4535a94c2b38d09912d7e8bab371c9e97be38
References: bsc#1050536 FATE#322898 bsc#1050538 FATE#322897

Fix compilation warning
qed_iwarp.c:1721:5: warning: ll2_syn_handle may be used
uninitialized in this function

Signed-off-by: Michal Kalderon <Michal.Kalderon@cavium.com>
Signed-off-by: Ariel Elior <Ariel.Elior@cavium.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/qlogic/qed/qed_iwarp.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/net/ethernet/qlogic/qed/qed_iwarp.c
+++ b/drivers/net/ethernet/qlogic/qed/qed_iwarp.c
@@ -1724,7 +1724,7 @@ qed_iwarp_ll2_comp_syn_pkt(void *cxt, st
 	int rc;
 
 	memset(&cm_info, 0, sizeof(cm_info));
-
+	ll2_syn_handle = p_hwfn->p_rdma_info->iwarp.ll2_syn_handle;
 	if (GET_FIELD(data->parse_flags,
 		      PARSING_AND_ERR_FLAGS_L4CHKSMWASCALCULATED) &&
 	    GET_FIELD(data->parse_flags, PARSING_AND_ERR_FLAGS_L4CHKSMERROR)) {
@@ -1740,7 +1740,6 @@ qed_iwarp_ll2_comp_syn_pkt(void *cxt, st
 		goto err;
 
 	/* Check if there is a listener for this 4-tuple+vlan */
-	ll2_syn_handle = p_hwfn->p_rdma_info->iwarp.ll2_syn_handle;
 	listener = qed_iwarp_get_listener(p_hwfn, &cm_info);
 	if (!listener) {
 		DP_VERBOSE(p_hwfn,
